### PR TITLE
global constants should have a prefix

### PR DESCRIPTION
**Pull request by sallyransom**
@iovortex @tLewisII @Cananito @yfujiki please review

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -34,11 +34,12 @@ Writing Objective-C? Check out our [Objective-C Style Guide](https://github.com/
 
 ## Naming
 
-Use descriptive names with camel case for classes, methods, variables, etc. Class names should be capitalized, while method names and variables should start with a lower case letter.
+Use descriptive names with camel case for classes, methods, variables, etc. Class names should be capitalized, while method names and variables should start with a lower case letter. Global constants, however, should have a project prefix.
 
 **Preferred:**
 
 ```swift
+let XYZSomeGlobalConstant = 123.456
```
Comment (Cananito) [inline README.markdown]: @sallyransom Devil's advocate + information gathering: isn't it the job of Swift's modules to avoid name collision? A.K.A. making prefixes unnecessary.

Comment (sallyransom) [inline README.markdown]: Yeah, I'm not entirely partial to this. But it keeps me lazy and especially helps avoid mistakes.

Comment (yfujiki) [inline README.markdown]: I was using global constant because Swift doesn't yet support class variable. 

But it seems we can either use 
- computed variable
- static let ...

in the class.

```
class A {
    class var constantPI: CGFloat {
        return 3.141592
    }

    static let constantMile = 1.60934
}
```

and we can access `ClassA.constantPI` or `ClassA.constantMile`. Probably we can promote this approach rather than global constant.

As far as I understand, Swift's module is not as accessible thing as in other languages. When you create a project, it will automatically package all the classes in the project modules so that naming convention in other projects will not conflict with your local convention. (i.e., you still need to deal with conflicts within your own project) I would be happy to be wrong :) This is one of the shortcomings in Swift IMO.

Comment (Cananito) [inline README.markdown]: I see. I'm completely ignorant when it comes to namespacing this kind of stuff. The standard library doesn't have any examples, and browsing the web a bit isn't that helpful.

I think enforcing global constants namespacing through structs (rather than clases) is a very elegant solution, but seems like a bit more of work.

Usage:

`Module.Struct.Constant` vs `ABCConstant`.

Usage wise, I would vote in favor of first option. Implementation wise, I would lean on the second.

I'll give a neutral vote: :+1: :-1:

Comment (yfujiki) [inline README.markdown]: Yeah, I don't think we have a clear winner. I am okay to leave it to dev's preference.

Comment (yfujiki) [inline README.markdown]: @sallyransom I have side-tracked, but I agree with @Cananito that we do not necessarily need `project prefix`. Project is already implicitly packaged into own module. (Yes, we are doing DSX prefix in our project, but IMO, that is rather to make it look aligned with other Objective-C constants).

Comment (Cananito) [inline README.markdown]: @sallyransom @yfujiki I have good news! Constants **ARE** namespaced. See this zipped project: https://drive.google.com/a/chaione.com/file/d/0B9jhxByanl4dUjBGY3c4ZVcyeDA/view?usp=sharing

Comment (Cananito) [inline README.markdown]: @sallyransom @yfujiki Because of this. I will change my vote to no prefixes and rely on modules/namespacing:

:-1:

Comment (Cananito) [inline README.markdown]: @sallyransom @yfujiki The detail I'm not sure about: `MyConstant` vs `myConstant`. I think I prefer the first one.

Comment (yfujiki) [inline README.markdown]: @Cananito Hahaa, nice experiment. Yes, that was along my understanding too. First and Second are the automatically generated modules and you do not need to do handle conflict between them.

I vote for `MyConstant` as well.

= You don't need to read the rest, but let me elaborate my argument for my own sake =
The stuff I was arguing by side tracking (global variable vs. name spacing) was conflict WITHIN the module, but that should be probably in another PR if needed.

In the following situation, `ViewAHeight` is declared as global constant so that it is accessible from outside of `ViewA`.  

```
let ViewAHeight = CGFloat(...)     // <=== Here

class ViewA: UIView {
    ...
}

class ViewControllerA: UIViewController {
    ...
    override func viewDidLoad() {
        super.viewDidLoad()
        ...
        view.height = ViewAHeight   // <== Here
        ...
    }
}    
```

But this can be written this way

```
class ViewA: UIView {
    static let height = CGFloat(...)     // <=== Here
    ...
}

class ViewControllerA: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        ...
        view.height = ViewA.height     // <=== Here
        ...
    }
}    
```

My preference as of now is to go with the latter, but I am not sure if we should enforce this in the style guide or not. Probably it is up to the developer. (If swift supported class variable, that is much easier vote...)

Comment (Cananito) [inline README.markdown]: @yfujiki Ah, I understand now. Yeah I think this one should be taken to another PR since it's independent of the prefix discussion.

**Comment by sallyransom** [comment]
@iovortex @tLewisII @Cananito @yfujiki please review again. this last commit shows updates from our previous discussions.

**Comment by Cananito** [comment]
@sallyransom Looks good to merge to me!
